lib/services/spatial.js

```diff
@@ -59,34 +152,35 @@ class SpatialService {
    * @return {[BBox]}              m e g a  b b o x
    */
   _expandBbox(layerResults) {
-    return layerResults.reduce((wholeBbox, {layer: {bbox: bbox}}) => {
+    return layerResults.reduce((wholeBbox, {
+      layer: {
+        bbox: bbox
+      }
+    }) => {
       return wholeBbox.merge(bbox);
     }, new BBox());
   }
 
 
-  _upsertLayers(req, res, core, datasets) {
-    var upsertSpecs = datasets.map(([layer, ds]) => [layer, ds.body.id]);
-    req.log.debug(`upsert specs ${upsertSpecs}`);
-
+  _upsertLayers(req, res, core, layers) {
     var fail = _.once((code, reason) => {
-      var fourfours = datasets.map(([_layer, ds]) => ds.body.id);
-      this._destroyLayers(fourfours, core, () => {
+      this._destroyLayers(layers, core, () => {
         res.status(code).send(reason);
       });
     });
+
     //this is a little different than the two preceding core calls. core.upsert
     //just sets up the request, so the callback will
     //return to the finish callback a list of *open* requests, rather than completed
     //requests, as well as the corresponding layer. Then we need to pipe the
     //layer's scratch file to the open request
-    return async.map(upsertSpecs, core.upsert.bind(core), (err, upsert) => {
+    return async.map(layers, core.upsert.bind(core), (err, upsert) => {
```

Comment:
Change to `async.mapLimit` instead of `async.map` as you have in other map calls?